src/app/controllers/collaborator.py

```diff
@@ -17,13 +17,28 @@ def get_all_collabs():
     mimetype="application/json"
   )
 
-@collabs.route("/", methods=["POST"])
+@collabs.route("/cadastro", methods=["POST"])
+@has_logged()
+@required_fields(['nome', 'genero', 'nascimento', 'telefone', 'bairro', 'cargo', 'cep', 'email', 'localidade', 'logradouro', 'numero', 'uf'])
 def insert_collabs():
     collabs = request.get_json()
-
-    mongo_client.collabs.insert_one(collabs)
-
-    return {"sucesso": f"Colaborador inserido com sucesso"}, 201
+    print(collabs)
+    allCollabs = list(mongo_client.collabs.find())
+  
+    exists = False
+    checkEmail = collabs['email']
+    for collab in allCollabs:
+      if collab['email'] == checkEmail:
+        exists = True
+    if not exists:
+      collabs['id'] = len(allCollabs) + 1
+      mongo_client.collabs.insert_one(collabs)
+      return {"sucesso": "Colaborador cadastrado com sucesso."}, 201
```

Comment:
Marcelo.... ali na endpoint post /cadastro.... aquele teste se existe o colaborador com email... sugiro escrever um decorator... ele vai ser identico a este aqui... só muda a collection ali e adapte... depois só decorar... e no corpo da enpoint pode botar direto a inserção do collab sem medo de ser feliz.

def user_exists():
    def _user_exists(function_current):
        @wraps(function_current)
        def __user_exists(*args, **kwargs):
            user = request.get_json()

            search = mongo_client.users.find_one({"email": user['email']})

            if not search:
                return function_current(*args, **kwargs)
            return jsonify({"error": f"Usuário {user['email']} já existe."})
        return __user_exists
    return _user_exists

Reply:
Testa esse aqui em baixo pra ver.
```
def collab_exists():
    def _collab_exists(function_current):
        @wraps(function_current)
        def __collab_exists(*args, **kwargs):
            collab = request.get_json()

            search = mongo_client.collaborators.find_one({"email": collab['email']})

            if not search:
                return function_current(*args, **kwargs)
            return jsonify({"error": f"Colaborador com email {user['email']} já existe."})
        return __collab_exists
    return _collab_exists
```

Reply:
Não daria para substituir esse "search" pelo decorator "user_exists"? Porque ele faz essa função de validação.

Reply:
Boa ideia, Maycon!